qualtran/surface_code/azure_cost_model.ipynb

```diff
@@ -0,0 +1,726 @@
+   "source": [
+    "# Azure Cost Model\n",
+    "In this notebook, we go through the cost model in \"Assessing requirements to scale to practical quantum advantage\" by [Beverland et al](https://arxiv.org/abs/2211.07629) to reproduce the estimated costs in table IV."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The Azure cost model has 5 inputs, these in no particular order are:\n",
+    "1. The specs of the circuit/algorithm to execute (e.g. number of T gates, ..etc).\n",
+    "1. The physical assumptions about the hardware (e.g. error rate, the latency of Clifford and measurement operations, ...etc).\n",
+    "1. The magic state factory (e.g. number of qubits consumed by it, its error rate, ...etc)\n",
+    "1. Cost of Approximating rotations up to error $\\epsilon$\n",
+    "1. The quantum error correction scheme.\n",
+    "\n",
+    "We take a look at each of these and then reproduce the results of appendix F."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## The Inputs\n",
+    "### Algorithm Specs\n",
+    "The first input to the estimation is a summary of the circuit or algorithm to execute. This summary is a couple of simple counts:\n",
+    "- `algorithm_qubits`: Number of algorithm qubits.\n",
+    "- `measurements`: Number of Clifford measurements ($M_R$).\n",
+    "- `t_gates`: Number of T gates ($M_T$).\n",
+    "- `toffoli_gates`: Number of Toffoli gates ($M_{Tof}$).\n",
+    "- `rotation_gates`: Number of Rotations ($M_R$).\n",
+    "- `rotation_circuit_depth`: Depth of rotation circuit  ($D_R$).\n",
+    "\n",
+    "Note: where we add the symbol used to refer to the quantity in the paper in parentheses.\n",
+    "\n",
+    "In Qualtran the algorithm specs are represented by the data class `qualtran.surface_code.class AlgorithmSpecs`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qualtran.surface_code import AlgorithmSpecs\n",
+    "\n",
+    "shor_specs = AlgorithmSpecs(\n",
+    "    algorithm_qubits=12581,\n",
+    "    rotation_gates=12,\n",
+    "    measurements=1.08e9,\n",
+    "    rotation_circuit_depth=12,\n",
+    "    toffoli_gates=3.73e10,\n",
+    "    t_gates=12,\n",
+    ")\n",
+    "shor_specs"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Physical Parameters\n",
+    "These are assumptions about the quantum hardware and are:\n",
+    "- `t_gate`: Latency of Clifford gates. \n",
+    "- `t_meas`: Latency of Measurement operations.\n",
+    "- `physical_error_rate`: Physical error rate ($p$)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from qualtran.surface_code.physical_parameters import BeverlandEtAl as physical_parameters\n",
```

Comment:
there's some weird pep8 things going on here. I think these physical parameter instantiations should be in all caps because they're a constant. I would try to avoid assigning them to a different name using "as". Use `=` assignment in a different line or (even better:) just use the explicit name